test/api_spec.js

```diff
@@ -953,6 +986,40 @@ describe("api", function () {
       expect(resource).not.to.be(null);
       expect(resource.bytes).to.eql(3381);
     }));
+    it('should restore a deleted resource by versions', function () {
+      this.timeout(TIMEOUT.LARGE);
+      return Q.all([
+        uploadImage({
```

Comment:
@strausr , shouldn't we upload 2 versions for the same publicId for checking this scenario?

Reply:
Based on the spec, it's a list that maps public_ids to versions. I don't think that you can map one public_id to many versions. That, being said, I'll check to see if that works. If it does, I'll add a separate test.

Reply:
I meant that for restoring an **old** version, you need to have at least 2 versions. The old and the new

Reply:
added a new test to restore an older version